src/code/audio/audio_thread.c

```diff
@@ -0,0 +1,977 @@
+            func_8018FA60(cmd->arg0, cmd->arg1, cmd->arg2, cmd->asInt);
+            break;
+
+        case AUDIOCMD_OP_GLOBAL_SET_REVERB_DATA:
+            AudioHeap_SetReverbData(cmd->arg1, cmd->arg0, cmd->asInt, false);
+            break;
+
+        default:
+            break;
+    }
+}
+
+void AudioThread_SetFadeOutTimer(s32 seqPlayerIndex, s32 fadeTimer) {
+    SequencePlayer* seqPlayer = &gAudioCtx.seqPlayers[seqPlayerIndex];
+
+    if (fadeTimer == 0) {
+        fadeTimer = 1;
+    }
+
+    seqPlayer->fadeVelocity = -(seqPlayer->fadeVolume / fadeTimer);
+    seqPlayer->state = SEQPLAYER_STATE_FADE_OUT;
+    seqPlayer->fadeTimer = fadeTimer;
+}
+
+void AudioThread_SetFadeInTimer(s32 seqPlayerIndex, s32 fadeTimer) {
+    SequencePlayer* seqPlayer;
+
+    if (fadeTimer != 0) {
+        seqPlayer = &gAudioCtx.seqPlayers[seqPlayerIndex];
+        seqPlayer->state = SEQPLAYER_STATE_FADE_IN;
+        seqPlayer->storedFadeTimer = fadeTimer;
+        seqPlayer->fadeTimer = fadeTimer;
+        seqPlayer->fadeVolume = 0.0f;
+        seqPlayer->fadeVelocity = 0.0f;
+    }
+}
+
+void AudioThread_InitMesgQueuesInternal(void) {
+    gAudioCtx.threadCmdWritePos = 0;
+    gAudioCtx.threadCmdReadPos = 0;
+    gAudioCtx.threadCmdQueueFinished = false;
+
+    gAudioCtx.taskStartQueueP = &gAudioCtx.taskStartQueue;
+    gAudioCtx.threadCmdProcQueueP = &gAudioCtx.threadCmdProcQueue;
+    gAudioCtx.audioResetQueueP = &gAudioCtx.audioResetQueue;
+
+    osCreateMesgQueue(gAudioCtx.taskStartQueueP, gAudioCtx.taskStartMsgs, ARRAY_COUNT(gAudioCtx.taskStartMsgs));
+    osCreateMesgQueue(gAudioCtx.threadCmdProcQueueP, gAudioCtx.threadCmdProcMsgBuf,
+                      ARRAY_COUNT(gAudioCtx.threadCmdProcMsgBuf));
+    osCreateMesgQueue(gAudioCtx.audioResetQueueP, gAudioCtx.audioResetMesgs, ARRAY_COUNT(gAudioCtx.audioResetMesgs));
+}
+
+void AudioThread_QueueCmd(u32 opArgs, void** data) {
+    AudioCmd* cmd = &gAudioCtx.threadCmdBuf[gAudioCtx.threadCmdWritePos & 0xFF];
+
+    cmd->opArgs = opArgs;
+    cmd->data = *data;
+
+    gAudioCtx.threadCmdWritePos++;
+
+    if (gAudioCtx.threadCmdWritePos == gAudioCtx.threadCmdReadPos) {
+        gAudioCtx.threadCmdWritePos--;
+    }
+}
+
+void AudioThread_QueueCmdF32(u32 opArgs, f32 data) {
+    AudioThread_QueueCmd(opArgs, (void**)&data);
+}
+
+void AudioThread_QueueCmdS32(u32 opArgs, s32 data) {
+    AudioThread_QueueCmd(opArgs, (void**)&data);
+}
+
+void AudioThread_QueueCmdS8(u32 opArgs, s8 data) {
+    u32 uData = data << 0x18;
+
+    AudioThread_QueueCmd(opArgs, (void**)&uData);
+}
+
+void AudioThread_QueueCmdU16(u32 opArgs, u16 data) {
+    u32 uData = data << 0x10;
+
+    AudioThread_QueueCmd(opArgs, (void**)&uData);
+}
+
+s32 AudioThread_ScheduleProcessCmds(void) {
+    static s32 D_801D5FF4 = 0;
+    s32 ret;
+
+    if (D_801D5FF4 < (u8)((gAudioCtx.threadCmdWritePos - gAudioCtx.threadCmdReadPos) + 0x100)) {
+        D_801D5FF4 = (u8)((gAudioCtx.threadCmdWritePos - gAudioCtx.threadCmdReadPos) + 0x100);
+    }
```

Comment:
Interesting, this seems useless (unless I'm missing something). Possibly a leftover debug variable?
Maybe `maxWriteReadDiff` or something similar?

Reply:
Yeah, it's completely unused and useless, the extra `0x100` offset is weird as well. I'm fine with that name